src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -204,8 +205,61 @@ def child_sort_key(item: TraceEvent) -> List[int]:
 
 
 def query_trace_data(
-    trace_id: str, params: Mapping[str, str]
+    trace_id: str, params: Mapping[str, str], use_snql: bool = False
 ) -> Tuple[Sequence[SnubaTransaction], Sequence[SnubaError]]:
+    if use_snql:
+        transaction_query = QueryBuilder(
+            Dataset.Discover,
+            params,
+            query=f"trace:{trace_id}",
+            selected_columns=[
+                "id",
+                "transaction.status",
+                "transaction.op",
+                "transaction.duration",
+                "transaction",
+                "timestamp",
+                "project",
+                "project.id",
+                "trace.span",
+                "trace.parent_span",
+                'to_other(trace.parent_span, "", 0, 1) AS root',
+            ],
+            # We want to guarantee at least getting the root, and hopefully events near it with timestamp
+            # id is just for consistent results
+            orderby=["-root", "timestamp", "id"],
+        )
+        error_query = QueryBuilder(
+            Dataset.Events,
+            params,
+            query=f"trace:{trace_id}",
+            selected_columns=[
+                "id",
+                "project",
+                "project.id",
+                "timestamp",
+                "trace.span",
+                "transaction",
+                "issue",
+                "title",
+                "tags[level]",
+            ],
+            # Don't add timestamp to this orderby as snuba will have to split the time range up and make multiple queries
+            orderby=["id"],
+            auto_fields=False,
+        )
+        results = bulk_snql_query(
+            [transaction_query.get_snql_query(), error_query.get_snql_query()],
+            referrer="api.trace-view.get-events",
```

Comment:
Feel free to ignore this but using the same referrer for each of the bulk queries has been a pet peeve of mine since the two queries are different but we have no way of distinguishing between the two in the transaction.

Reply:
agreed, but the current bulk APIs don't support a way to provide >1 referrer yet. I have this mentally backlogged but let me create something in Jira